### PR TITLE
connect with sdk react-native components

### PR DESCRIPTION
**Pull request by hvhallmann**
# Description

Include support to @sonr-io/react-native-ui-components instead of current Login flow

# Impact

- Impact: (Critical / High / Medium / Low)
  - Critical: Architectural changes / big refactorings / changes that touches core parts that can impact service availability
Needs to test the login process agagin

# How to test

Create one account and use .env Users to set matrix username and password

Is there any setup steps needed? no
Is there unit tests to cover the new changes? no

## 1. (Example) User should be able to see printed value

- Go to create account button
- Insert username and password, click next
- Insert matrixUsername and password.
- The system should login you directly if key match

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

**Comment by hvhallmann** [comment]
please check if it is working in your local machine,
I have the feeling that one package is broken

**Comment by hvhallmann** [comment]
The compiler is complaining about unmeet dependency with react-native-ui-components.
I tried to update ours, but got into a bunch of failures,
Let me know if this is required to merge

Screen Shot 2022-07-25 at 18 43 06

File: src/screens/Splash.tsx [hunk 1]
```diff
@@ -1,24 +1,45 @@
...
 import { loginWithSession } from "../lib/matrix";
+import { AuthenticationComponent } from "@sonr-io/react-native-ui-components";
 
 type Props = StackScreenProps<StackParams, "Login">;
 
 const SplashScreen: React.FC<Props> = ({ navigation }) => {
+  const [showAuthentication, setShowAuthentication] = useState(false);
+  const [loading, setLoading] = React.useState(false);
+
+  const onSuccessHandler = async (data: any) => {
```
Comment (tgfassina) [inline src/screens/Splash.tsx:27]: Here we can use `SonrUserData` instead of `any`

File: src/screens/Splash.tsx [hunk 2]
```diff
@@ -1,24 +1,45 @@
 import { StackScreenProps } from "@react-navigation/stack";
-import React, { useEffect } from "react";
-import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
+import React, { useEffect, useState } from "react";
+import {
+  ActivityIndicator,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
```
Comment (tgfassina) [inline src/screens/Splash.tsx:7]: Can this go away? I think it's here only because we made the logo clickable, but that's no longer the case

**Review by tgfassina — CHANGES_REQUESTED** [review]
It authenticates nicely, but one feature that we broke was the session login, meaning that I have to log in every time I restart the app. If I log in and then restart the app, I shouldn't need to log in again.

Also this is probably the best time to delete the legacy login screen code.

File: src/screens/Splash.tsx (hunk 2 above)
Comment (hvhallmann) [reply src/screens/Splash.tsx:7]: yep, good point is no longer necessary

File: src/screens/Splash.tsx (hunk 1 above)
Comment (hvhallmann) [reply src/screens/Splash.tsx:27]: good point!

Comment (hvhallmann) [reply src/screens/Splash.tsx:27]: I forgot to save the session once the user logs in.
Now it is fixed

**Review by tgfassina — APPROVED** [review]
@hvhallmann really nice! Good to finally see this things working together 👏🏻

**Pull request merged**